docs/guide/search/search_criteria_reference.md

```diff
@@ -0,0 +1,55 @@
+# Search reference
+
+Search Criteria are filters for Content and Location Search.
+
+A Criterion consist of two parts:
+
+- The API Value: `Criterion`
+- Specific handler per search engine: `CriterionHandler`
+
+`Criterion` represents the value you use in the API, while `CriterionHandler` deals with the business logic in the background translating the value to something the search engine can understand. `CriterionHandler` also handles `ezkeyword` external storage for Legacy (SQL-based) search.
```

Comment:
CriterionHandler for every FT having ext storage should take it into the account, so for me too much detail here.